src/paint_canvas.rs

```diff
@@ -145,57 +122,13 @@ impl PaintCanvas {
       }
    }
 
-   /// Updates chunks that have been decoded between the last call to `update` and the current one.
-   pub fn update(&mut self, renderer: &mut Backend) {
-      while let Ok((chunk_position, image)) = self.decoded_chunks_rx.try_recv() {
-         let chunk = self.ensure_chunk(renderer, chunk_position);
-         chunk.upload_image(&image, (0, 0));
-      }
-      while let Ok((chunk_position, image)) = self.encoded_chunks_rx.try_recv() {
-         let chunk = self.ensure_chunk(renderer, chunk_position);
-         chunk.image_cache = Some(image);
-         self.chunk_cache_timers.insert(chunk_position, Instant::now());
-      }
-      for (chunk_position, instant) in &self.chunk_cache_timers {
-         if instant.elapsed() > Self::CHUNK_CACHE_DURATION {
-            if let Some(chunk) = self.chunks.get_mut(chunk_position) {
-               chunk.image_cache = None;
-            }
-         }
-      }
-   }
-
-   /// Returns a receiver for the image data of the chunk at the given position, if it's not empty.
-   ///
-   /// The chunk data that arrives from the receiver may be `None` if encoding failed.
-   pub fn enqueue_network_data_encoding(
+   pub fn set_chunk(
       &mut self,
-      output_channel: mpsc::UnboundedSender<((i32, i32), ChunkImage)>,
-      chunk_position: (i32, i32),
+      renderer: &mut Backend,
+      (chunk_position, image): ((i32, i32), RgbaImage),
```

Comment:
Is there a reason for making this into a tuple `((i32, i32), RgbaImage)`, instead of keeping it as two separate arguments?